File: parcels/user/src/modals/reset.js
import "../../use";
import { user } from "../user";

const { component } = await use("@/rollo/");
const { Form, Input } = await use("@/form/");
const { modal } = await use("@/modal/");
const { Spinner } = await use("/tools/spinner");
const { Alert } = await use("/tools/alert");

export const Reset = async () => {
  const form = Form(
    `flex flex-col gap-y-3 py-1`,
    { "[reset]": true },
    Input({
      type: "email",
      label: "Email",
      name: "email",
      required: true,
    }),
  );

  const submit = component.button(".btn.btn-primary", {
    type: "button",
    text: "Send",
    disabled: true,
  });

  form.$.effects.add(
    ({ valid }, message) => {
      submit.disabled = !valid;
    },
    ["valid"],
  );

  const content = (host) => {
    host.tree.content.attribute.user = true;
    //console.log("host.tree:", host.tree); ////
    submit.on.click(async (event) => {
      form.clear(".alert");
      const valid = form.valid;
      if (valid) {
        const { email } = form.data;

        const data = await Spinner.while(
          {
            parent: host.tree.content,
            size: "6rem",
          },
          async () => {
            return await user.reset(email);
          },
        );

        //console.log("data:", data); ////
        if (data.error) {
          form.append(Alert(data.error, { style: "danger" }));
        } else {
          document.body.focus(); ////
          submit.remove();
          form.append(
            Alert(`An email with a reset link has been sent`, {
              style: "success",
            }),
          );
        }
      }
    });
    return form;
  };
  const result = await modal({ content, title: "Reset password" }, submit);
  return result;
};

File: _history/20260206/test/client_code/tests/modal/reset.py
"""
modal/reset.py
"""


def main(use):

    send_password_reset_email = use.anvil.send_password_reset_email

    print("send_password_reset_email:", send_password_reset_email)

    Form = use("@/form/").Form
    Input = use("@/form/").Input
    Alert = use("/tools/alert").Alert

    modal = use("@@/modal/")
    component = use("@@/component/")
    reset = use("@@/user:reset")
    ##print("reset:", reset)  ##

    def content(close):

        submit = component.button(
            ".btn.btn-primary", type="button", text="Send", disabled=True
        )

        message = component.div()

        form = Form(
            ".d-flex.flex-column.gap-3",
            Input(dict(type="email", name="email", label="Email", required=True)),
            message,
            component.menu(
                ".d-flex.justify-content-end",
                submit,
            ),
        )

        def effect(change, message):
            """."""
            ##print("change", change)  ##
            valid = change.valid
            submit.disabled = not valid

        form["$"].effects.add(effect, ["valid"])

        @submit.on()
        def click(event):
            message.clear()
            if form.valid:
                email = form.data.email
                response = reset(email)
                if response.get('ok'):
                    submit.remove()
                    message.append(Alert('An email with a reset link has been sent', dict(style='success')))
                else:
                    message.append(Alert(response.get('message', ''), dict(style='danger')))

            ##close(True)

        return form

    result = modal(content, title="Reset password")
    print("result:", result)
